### DISCUSSION TITLE
Reflection scan for annotation in native image

### DISCUSSION THREAD

**Discussion by markrouz** [quarkusio/quarkus#29859] [Q&A] [answered]
I try to use library https://github.com/ronmamo/reflections in my quarkus app.
In particular I use method 
`org.reflections.Reflections#getTypesAnnotatedWith(java.lang.Class<? extends java.lang.annotation.Annotation>, boolean)`
in dev mode this method returns correct set of classes with specified annotation (this annotation has runtime retention policy), but in native mode (I mean, for example, when I launch tests with @QuarkusIntegrationTest) it returns nothing.
Could you tell me please is it possible to scan classes for specified annotation in native mode?

**Comment by maxandersen** [comment, marked as answer]
Native mode has limitations - one of them is arbitrary open ended reflection.

Look at https://quarkus.io/guides/writing-native-applications-tips#registering-for-reflection how you might be able to have it working. 

In general though we recommend doing such scanning at buildtime as an extensions rather than do it via operations that only works in jvm mode.
